components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -583,6 +645,43 @@ class PlacesWriterConnection internal constructor(connHandle: Long, api: PlacesA
         }
     }
 
+    override suspend fun addHistoryMetadata(metadata: HistoryMetadata): String {
+        val builder = MsgTypes.HistoryMetadata.newBuilder()
```

Comment:
@csadilek found that this isn't friendly to our nullable values; should probably convert to defaults (e.g. empty strings) if need be.